File: typescript/10-audio.ts
/**
 * A1 speech + transcription, and Gemini token audio on Interactions.
 *
 *   export TONIA_API_KEY=tonia_sk_…
 *   npx tsx 10-audio.ts
 *
 * Sold A1 examples: mistral/voxtral-mini-tts-2603, openai/gpt-transcribe.
 * Gemini token TTS/STT uses interactions.create.
 * Workspace audio must be on (portal /dlp).
 */
import { Tonia } from "@tonia-router/sdk";

const BARE_FISH = new Set(["s2.1-pro", "s2.1-pro-free"]);

function skip(id: string): boolean {
  const lowered = id.toLowerCase();
  if (BARE_FISH.has(lowered)) return true;
  return /-latest|vd-|realtime/i.test(id);
}

function tinyWav(): Uint8Array {
  return Uint8Array.from([
    0x52, 0x49, 0x46, 0x46, 0x24, 0x00, 0x00, 0x00, 0x57, 0x41, 0x56, 0x45,
    0x66, 0x6d, 0x74, 0x20, 0x10, 0x00, 0x00, 0x00, 0x01, 0x00, 0x01, 0x00,
    0x40, 0x1f, 0x00, 0x00, 0x80, 0x3e, 0x00, 0x00, 0x02, 0x00, 0x10, 0x00,
    0x64, 0x61, 0x74, 0x61, 0x00, 0x00, 0x00, 0x00,
  ]);
}

const client = new Tonia({
  apiKey: process.env.TONIA_API_KEY,
  defaultHeaders: {
    "HTTP-Referer": "https://github.com/tonia-router/sdk-examples",
    "X-Tonia-Title": "sdk-examples",
  },
});

const { data } = await client.models.list();
const rows = data.filter((item) => !skip(item.id));
const caps = (item: (typeof rows)[number]) => item.capabilities ?? [];
const surfaceOf = (item: (typeof rows)[number]) => item.surface ?? {};

const tts = rows.find(
  (item) =>
    (surfaceOf(item).path === "/v1/audio/speech" ||
      caps(item).includes("audio_speech")) &&
    !item.id.startsWith("gemini/"),
)?.id;
const stt = rows.find(
  (item) =>
    (surfaceOf(item).path === "/v1/audio/transcriptions" ||
      caps(item).includes("audio_transcription")) &&
    !item.id.startsWith("gemini/"),
)?.id;
const geminiAudio = rows.find(
  (item) =>
    item.id.startsWith("gemini/") &&
    (surfaceOf(item).family === "speech" ||
      surfaceOf(item).family === "transcription"),
)?.id;

if (!tts && !stt && !geminiAudio) {
  throw new Error("no listed audio on this key");
}

const spent: Record<string, unknown> = {};
if (tts) {
  const speech = await client.audio.speech.create({
    model: tts,
    input: "Bonjour Tonia",
    voice: tts.startsWith("fish_") || tts.includes("s2.1-pro") ? "" : "alloy",
  });
  spent.speech = {
    model: tts,
    kind: speech instanceof Uint8Array ? "bytes" : "json",
  };
}
if (stt) {
  const transcript = await client.audio.transcriptions.create({
    model: stt,
    file: tinyWav(),
    filename: "clip.wav",
  });
  spent.transcription = { model: stt, has_text: Boolean(transcript && typeof transcript === "object") };
}
if (geminiAudio) {
  await client.interactions.create({
    model: geminiAudio,
    input: "Bonjour",
    stream: false,
  });
  spent.gemini_interactions = { model: geminiAudio };
}
console.log(JSON.stringify(spent, null, 2));

File: python/10_audio.py
"""A1 speech + transcription, and Gemini token audio on Interactions.

    export TONIA_API_KEY=tonia_sk_…
    python 10_audio.py

Sold A1 examples: mistral/voxtral-mini-tts-2603, openai/gpt-transcribe.
Gemini token TTS/STT uses interactions.create.
Workspace audio must be on (portal /dlp).
"""

from __future__ import annotations

import json
import os
import struct
import wave
from io import BytesIO

from tonia import Tonia

_BARE_FISH = frozenset({"s2.1-pro", "s2.1-pro-free"})


def _skip(model_id: str) -> bool:
    lowered = model_id.lower()
    if lowered in _BARE_FISH:
        return True
    return any(part in lowered for part in ("-latest", "vd-", "realtime"))


def _caps(item: dict[str, object]) -> list[str]:
    raw = item.get("capabilities")
    return [str(cap) for cap in raw] if isinstance(raw, list) else []


def _surface(item: dict[str, object]) -> dict[str, object]:
    raw = item.get("surface")
    return raw if isinstance(raw, dict) else {}


def _tiny_wav() -> bytes:
    buf = BytesIO()
    with wave.open(buf, "wb") as out:
        out.setnchannels(1)
        out.setsampwidth(2)
        out.setframerate(8000)
        out.writeframes(struct.pack("<64h", *([0] * 64)))
    return buf.getvalue()


with Tonia(
    api_key=os.environ["TONIA_API_KEY"],
    default_headers={
        "HTTP-Referer": "https://github.com/tonia-router/sdk-examples",
        "X-Tonia-Title": "sdk-examples",
    },
) as client:
    listed = client.models.list()
    rows = [item for item in listed["data"] if not _skip(item["id"])]

    tts = next(
        (
            item["id"]
            for item in rows
            if (
                _surface(item).get("path") == "/v1/audio/speech"
                or "audio_speech" in _caps(item)
            )
            and not item["id"].startswith("gemini/")
        ),
        None,
    )
    stt = next(
        (
            item["id"]
            for item in rows
            if (
                _surface(item).get("path") == "/v1/audio/transcriptions"
                or "audio_transcription" in _caps(item)
            )
            and not item["id"].startswith("gemini/")
        ),
        None,
    )
    gemini_audio = next(
        (
            item["id"]
            for item in rows
            if item["id"].startswith("gemini/")
            and _surface(item).get("family") in ("speech", "transcription")
        ),
        None,
    )
    if not tts and not stt and not gemini_audio:
        raise SystemExit("no listed audio on this key")

    spent: dict[str, object] = {}
    if tts:
        speech_voice = "" if tts.startswith("fish_") or "s2.1-pro" in tts else "alloy"
        speech = client.audio.speech.create(model=tts, input="Bonjour Tonia", voice=speech_voice)
        spent["speech"] = {
            "model": tts,
            "kind": "bytes" if isinstance(speech, (bytes, bytearray)) else "json",
        }
    if stt:
        transcript = client.audio.transcriptions.create(
            model=stt,
            file=_tiny_wav(),
            filename="clip.wav",
        )
        spent["transcription"] = {"model": stt, "has_text": isinstance(transcript, dict)}
    if gemini_audio:
        client.interactions.create(model=gemini_audio, input="Bonjour", stream=False)
        spent["gemini_interactions"] = {"model": gemini_audio}
    print(json.dumps(spent, indent=2))
